src/os/bluestore/KernelDevice.cc

```diff
@@ -366,7 +366,13 @@ void KernelDevice::_aio_thread()
 		 << " ioc " << ioc
 		 << " with " << (ioc->num_running.load() - 1)
 		 << " aios left" << dendl;
-	assert(r >= 0);
+	if (r < 0) {
+	  derr << __func__ << " finished aio " << aio[i] << " r " << r
+	       << " ioc " << ioc
+	       << " with " << (ioc->num_running.load() - 1)
+	       << " aios left" << dendl;
+	  assert(r >= 0);
+    }
 
```

Comment:
This prints 2 times when `r < 0` right ? Please see https://github.com/mychoxin/ceph/blob/deb79f35fde7c6c49083685673562e4fe18bb506/src/os/bluestore/KernelDevice.cc#L365